### PR TITLE
feat(native): wrap common NSObject functions in an OCaml context

### PR DESCRIPTION
**Pull request by zbaylin**
This creates a basic wrapper of various `NSObject` selectors, such as `isEqual`, `hash`, `description`, etc.

This will be helpful in the creation of native menus on macOS, since we'll have generic functions for comparing values, which will be useful in data structures like `Hashtbl.t`

### CODE REVIEW

File: src/Native/NSObject.re [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+%ifdef
+USE_COCOA;
+
+type t('a) = 'a;
+
+let wrap: 'a => t('a) = nsObj => nsObj;
+
+/* equal
+   Uses the NSObject isEqual selector */
+external equal: (t('a), t('b)) => bool = "revery_NSObject_equal";
```
Comment (bryphe) [inline src/Native/NSObject.re:13]: Nice! Interface looks great.

File: src/Native/ReveryNSObject.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#import <objc/objc-runtime.h>
+
+#include <caml/alloc.h>
+#include <caml/callback.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+CAMLprim value revery_NSObject_equal(value vNSObjA, value vNSObjB) {
+    CAMLparam2(vNSObjA, vNSObjB);
+    NSObject *nsObjA = (NSObject *)vNSObjA;
```
Comment (bryphe) [inline src/Native/ReveryNSObject.c:12]: Should we wrap these pointers here? Just so we don't add another case we need to handle here: https://github.com/revery-ui/revery/pull/992

Comment (zbaylin) [reply src/Native/ReveryNSObject.c:12]: My plan was to get #992 merged into master then merge master into this. That PR includes fixes to the pointer wrap functions so I think it makes most sense to do it in that order.

Comment (bryphe) [reply src/Native/ReveryNSObject.c:12]: Makes sense!

**Review by bryphe — APPROVED** [review]
Looks great :+1:

**Pull request merged**